Document/0x05d-Testing-Data-Storage.md

```diff
@@ -251,54 +251,81 @@ if(Java.available){
+
+#### Scoped Storage
+
+To give users more control over their files and to limit file clutter, apps that target Android 10 (API level 29) and higher are given scoped access into external storage, or [scoped storage](https://developer.android.com/training/data-storage#scoped-storage), by default. When scoped storage is enabled, apps cannot access the app-specific directories that belong to other apps.
+
+The Android developers documentation provides a detailed guide highlighting common [storage use cases and best practices](https://developer.android.com/training/data-storage/use-cases) differentiating between handling media and non-media files and considering scoped storage.
+
+**Opting out**: Apps targeting Android 10 (API level 29) or lower can [temporarily opt out of scoped storage](https://developer.android.com/training/data-storage/use-cases#opt-out-in-production-app) using `android:requestLegacyExternalStorage="true"` in their app manifest. Once the app targets Android 11 (API level 30), the system ignores the `requestLegacyExternalStorage` attribute when running on Android 11 devices.
+
+> [App attribution for media files (Android Developers)](https://developer.android.com/training/data-storage/shared/media#app-attribution):
+> When [scoped storage](https://developer.android.com/training/data-storage#scoped-storage) is enabled for an app that targets Android 10 or higher, the system attributes an app to each media file, which determines the files that your app can access when it hasn't requested any storage permissions. Each file can be attributed to only one app. Therefore, if your app creates a media file that's stored in the photos, videos, or audio files media collection, your app has access to the file.
+>
+> If the user uninstalls and reinstalls your app, however, you must request [READ_EXTERNAL_STORAGE](https://developer.android.com/reference/android/Manifest.permission#READ_EXTERNAL_STORAGE) to access the files that your app originally created. This permission request is required because the system considers the file to be attributed to the previously installed version of the app, rather than the newly installed one.
+
+For example, trying to access a file stored using the `MediaStore` API with a `content://` URI like `content://media/external_primary` would only work as long as the image _belongs_ to the invoking app (due to `owner_package_name` attribute in the `MediaStore`). If the app calls a `content://` URI that does not belong to the app, it will fail with a `SecurityException`:
```

Comment:
Good example. I can't tell for Android devs but wouldn't an example with `getExternalFilesDir` be easier here?
```

Reply:
Please post a suggestion with that case as well, we can have both but wanted to definitely highlight this

Reply:
I wanted to suggest the following:

```suggestion
For example, if you create a file within the external storage in the following way:

```kotlin
val path = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES);
val file = File(path, "image.jpg");
FileOutputStream(file).use { 
  bitmap.compress(Bitmap.CompressFormat.JPEG, 100, it) 
}


Accessing this image would only work as long as the image belongs to the invoking app.
```


But now when I think of it, the app would require WRITE_EXTERNAL_STORAGE, which isn't possible anymore. I believe that the current example fits just right.